### PR TITLE
linted visualization module

### PR DESCRIPTION
**Pull request by andrewelamb**
- fixes [fds-367](https://sagebionetworks.jira.com/browse/FDS-367)
- lints most of visualization module (save for some overly complex code that needs to be refactored)
- fixes some spelling 
- adds some type hinting

### CODE REVIEW

File: schematic/visualization/tangled_tree.py [hunk 1]
```diff
@@ -784,35 +864,37 @@ def get_ancestors_nodes(self, subgraph, components):
...
 
         if self.figure_type == "component":
             # Gather all source nodes
             source_nodes = self.find_source_nodes(nodes, edges)
 
             # Map all children to their parents and vice versa
-            child_parents, parent_children = self.get_parent_child_dictionary(
-                nodes, edges
-            )
+            child_parents, parent_children = self.get_parent_child_dictionary(edges)
```
Comment (mialy-defelice) [inline schematic/visualization/tangled_tree.py:892]: curious about how these args were passed before then now. To be safe lets set this as `self.get_parent_child_dictionary(edges=edges)`

**Review by mialy-defelice — CHANGES_REQUESTED** [review]

File: schematic/visualization/tangled_tree.py [hunk 2]
```diff
@@ -614,41 +665,53 @@ def adjust_node_placement(
...
-        for i, layer in enumerate(node_layers):
+        for _, layer in enumerate(node_layers):
             nodes_to_move = []
             for node in layer:
                 if node in source_nodes:
                     nodes_to_move.append(node)
             for node in nodes_to_move:
-                node_layers[i].remove(node)
-                node_layers[i].append(node)
+                layer.remove(node)
```
[the remark this answers is not in the document]
Comment (andrewelamb) [reply schematic/visualization/tangled_tree.py:685]: The two statements are equivelent
`node_layers[i].remove()` vs `layer.remove()`. The code is iterating through the layers so it's weird to use `layers[i]`

[the remark this answers is not in the document]
Comment (andrewelamb) [reply schematic/visualization/tangled_tree.py:685]:
```
list_list = [["a"], ["b"], ["c"]]
for layer in list_list:
    layer.remove("a")
print(list_list)
>>> [[], ['b'], ['c']]
```

File: schematic/visualization/tangled_tree.py [hunk 3]
```diff
@@ -614,41 +665,53 @@ def adjust_node_placement(
...
                 key: parent
                 value: list of the parents children
         Outputs:
-            layers_list (List(list): list of lists of dictionaries that specifies each node and its parents (if applicable)
+            layers_list (List(list): list of lists of dictionaries that specifies each node and its
+             parents (if applicable)
         """
         num_layers = len(node_layers)
-        layers_list = [[] for i in range(0, num_layers)]
+        layers_list: list[list[dict[str, list[str]]]] = [
```
[the remark this answers is not in the document]
Comment (andrewelamb) [reply schematic/visualization/tangled_tree.py:712]: So this is a thign about typing, when you introduce a new variable that's not an argument mypy will attempt to figure oput what it by the type of the thign you first assign it. That can be either too narrow or too broad. When it's not exactly what you need it to be you can assign it's type like you would an argument to a function/method.

[the remark this answers is not in the document]
Comment (andrewelamb) [reply schematic/visualization/tangled_tree.py:712]: If I don't type it here mypy just thinks its a list[list] which is too broad.

File: schematic/visualization/tangled_tree.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mialy-defelice) [reply schematic/visualization/tangled_tree.py:685]: Interesting, I didn't know that layer was still attached to node_layers...

File: schematic/visualization/tangled_tree.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mialy-defelice) [reply schematic/visualization/tangled_tree.py:712]: Okay that's good to know for the future.

**Review by mialy-defelice — APPROVED** [review]
I tested the API and CLI, all looks good!

**Pull request merged**